pyigm/metallicity/mcmc.py

```diff
@@ -0,0 +1,887 @@
+            plt.savefig('debug_inter_b_{}.png'.format(ii))
+            plt.clf()
+
+            #plane 3
+            original_3=self.mod_colm[ii][0,0,:,:]
+            imgplt=plt.imshow(original_3,origin='lower')
+            imgplt.set_clim(7.0,12.0)
+            plt.savefig('debug_orig_c_{}.png'.format(ii))
+            plt.clf()
+
+
+            interp_3=np.zeros((int_met.size,int_dens.size))
+            xx=0
+            for cc in int_met:
+                yy=0
+                for rr in int_dens:
+                    interp_3[xx,yy]=self.interpol[ii]([self.mod_axisval[0][0],self.mod_axisval[1][0],cc,rr])
+                    yy=yy+1
+                xx=xx+1
+            imgplt=plt.imshow(interp_3,origin='lower')
+            imgplt.set_clim(7.0,12.0)
+            plt.savefig('debug_inter_c_{}.png'.format(ii))
+            plt.clf()
+
+            #plane 4
+            original_4=self.mod_colm[ii][:,0,0,:]
+            imgplt=plt.imshow(original_4,origin='lower')
+            imgplt.set_clim(7.0,12.0)
+            plt.savefig('debug_orig_d_{}.png'.format(ii))
+            plt.clf()
+
+            interp_4=np.zeros((int_colm.size,int_dens.size))
+            xx=0
+            for cc in int_colm:
+                yy=0
+                for rr in int_dens:
+                    interp_4[xx,yy]=self.interpol[ii]([cc,self.mod_axisval[1][0],self.mod_axisval[2][0],rr])
+                    yy=yy+1
+                xx=xx+1
+            imgplt=plt.imshow(interp_4,origin='lower')
+            imgplt.set_clim(7.0,12.0)
+            plt.savefig('debug_inter_d_{}.png'.format(ii))
+            plt.clf()
+
+        return
+
+    def __call__(self, param):
+        """ This is the method that hides the computation of the likelihood/prior behind a callable function
+
+        Parameters
+        ----------
+        param
+
+        Returns
+        -------
+
+        """
+        #call the likelihood function [log]
+        lh = self.lnlhood(param)
+
+        #call the prior function [log]
+        lp = self.lnprior(param)
+
+        #return likelihood*prior [lor] or inf
+        if not np.isfinite(lh) or not np.isfinite(lp):
+            return -np.inf
+        else:
+            return lh + lp
+
+
+def mcmc_ions(data,infodata,model,nwalkers=500,nsamp=250,threads=12,
+              outsave='emceeout', optim=False, effnhi=True):
+    """ This is the main, which does not do much
+
+    Parameters
+    ----------
+    data : list
+      List of tuples containing the column density constraints
+    infodata : dict
+      dict containing observation info, e.g. NHI, z
+    model : str
+      Name of the grid
+    nwalkers : int, optional
+    nsamp : int, optional
+    threads : int, optional
+    outsave : str, optional
+    optim : bool, optional
+      Optimize?  [Not recommended]
+    effnhi : bool, optional
+
+    Returns
+    -------
+
+    """
+
+    print('Getting started...')
+
+    #Do not run with less than 100 samples
+    nsamp=np.max([nsamp,100])
+
```

Comment:
Given the call below in pyigm_mtlmcmc.py, we should perhaps add a floor to nwalkers as well. Something like    nwalkers=np.max([nwalkers,400])